specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2019-05-13/bms.json

```diff
@@ -0,0 +1,3352 @@
+    },
+    "AzureWorkloadSQLRestoreRequest": {
+      "description": "AzureWorkload SQL -specific restore. Specifically for full/diff restore",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AzureWorkloadRestoreRequest"
+        }
+      ],
+      "properties": {
+        "shouldUseAlternateTargetLocation": {
+          "description": "Default option set to true. If this is set to false, alternate data directory must be provided",
+          "type": "boolean"
+        },
+        "isNonRecoverable": {
+          "description": "SQL specific property where user can chose to set no-recovery when restore operation is tried",
+          "type": "boolean"
+        },
+        "alternateDirectoryPaths": {
+          "description": "Data directory details",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SQLDataDirectoryMapping"
+          }
+        }
+      },
+      "x-ms-discriminator-value": "AzureWorkloadSQLRestoreRequest"
+    },
+    "BEKDetails": {
+      "description": "BEK is bitlocker encryption key.",
+      "type": "object",
+      "properties": {
+        "secretUrl": {
+          "description": "Secret is BEK.",
+          "type": "string"
+        },
+        "secretVaultId": {
+          "description": "ID of the Key Vault where this Secret is stored.",
+          "type": "string"
+        },
+        "secretData": {
+          "description": "BEK data.",
+          "type": "string"
+        }
+      }
+    },
+    "BMSRPQueryObject": {
+      "description": "Filters to list backup copies.",
+      "type": "object",
+      "properties": {
+        "startDate": {
+          "format": "date-time",
+          "description": "Backup copies created after this time.",
+          "type": "string"
+        },
+        "endDate": {
+          "format": "date-time",
+          "description": "Backup copies created before this time.",
+          "type": "string"
+        },
+        "restorePointQueryType": {
+          "description": "RestorePoint type",
+          "enum": [
+            "Invalid",
+            "Full",
+            "Log",
+            "Differential",
+            "FullAndDifferential",
+            "All"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "RestorePointQueryType",
+            "modelAsString": true
+          }
+        },
+        "extendedInfo": {
+          "description": "In Get Recovery Point, it tells whether extended information about recovery point is asked.",
+          "type": "boolean"
+        }
+      }
+    },
+    "DiskExclusionProperties": {
+      "type": "object",
+      "properties": {
+        "diskLunList": {
+          "description": "List of Disks' Logical Unit Numbers (LUN) to be used for VM Protection.",
+          "type": "array",
+          "items": {
+            "format": "int32",
+            "type": "integer"
+          }
+        },
+        "isInclusionList": {
```

Comment:
We try to stay away from booleans in ARM as they aren't extensible. Can we model this as "InclusionList" with possible enum values "included" or "excluded"? I see most of the properties on these resources are defined as "is{something}". If we choose not to make this change for consistency purposes, consider collapsing related flags into single enums to simplify the model in future api versions.

Reply:
DiskExclusionProperties is to take input on what disks need to be excluded or included form the backup. diskLunList is the list of Disk LUN numbers and isInclusionList specifies whether the diskLunList is to be treated as an inclusion list or exclusion list. There is no scope of extending this field as the list can only be either inclusion or exclusion, there will never be a third kind of list. So, we for sure know that we are not going to extend this filed in the future.